scroll-search.lic

```diff
@@ -0,0 +1,533 @@
+
+	  if @owned_scrolls.key? (container)
+		my_scroll_list = @owned_scrolls[container]
+	  end
+
+	  my_scroll_list.push(scroll_info)
+	  if my_scroll_list.length > 0
+		echo("Adding scroll #{scroll_info.spell} to container #{container}. Length of array is #{my_scroll_list.length}") if $debug_mode
+		# Re-save owned scrolls listing
+		@owned_scrolls[container] = my_scroll_list
+			
+		# Re-save user variables
+		UserVars.owned_scrolls = @owned_scrolls
+			
+		# List scrolls
+		my_scroll_list.each do |scroll_array|
+		  echo("Scrolls in container is #{scroll_array.spell}") if $debug_mode
+		end
+			
+		# List everything
+		@owned_scrolls.each do |container, scroll_array|
+		  scroll_array.each do |scroll_information|
+			echo("Container #{container} | #{scroll_information.spell}") if $debug_mode
+		  end
+		end
+	  else
+		echo_message("COULD NOT ADD SCROLL TO CONTAINER")
+		exit
+	  end
+	end
+  end
+  
+  def grab_scroll(index)
+	echo ("Index in grab scroll of scroll is: #{index}.") if $debug_mode
+	
+    # Retrieve scroll info 
+	scroll_info = get_scroll_info(index)
+	echo("Got scroll info with spell: #{scroll_info.spell}") if $debug_mode
+
+	# Get previous values
+	cur_scroll_container = get_noun(scroll_info.container)
+	cur_scroll_ordinal = scroll_info.ordinal
+	cur_scroll_type = scroll_info.get_type
+
+	bput("open my #{cur_scroll_container}", 'That is already open', 'You open')
+	bput("get #{cur_scroll_ordinal} #{cur_scroll_type} from my #{cur_scroll_container}", 'You get')
+	
+	# Reorder container
+	reorder_container(scroll_info.container, scroll_info, true)
+	
+	# Remove this scroll from our hash map
+	scrolls_in_container = @owned_scrolls[scroll_info.container]
+	
+	echo("Scrolls in container count is: #{scrolls_in_container.length}") if $debug_mode
+	scrolls_in_container.each do |scroll_information|
+	  echo ("Index of current scroll in container is: #{scroll_information.index}. Index is: #{index}") if $debug_mode
+	  if scroll_information.index != index.to_i
+		next
+	  else
+		echo ("Found scroll we want to remove from listing... #{scroll_information.spell}") if $debug_mode
+		scrolls_in_container.delete(scroll_information)
+	  end
+		
+	  # Re-save owned scrolls listing
+	  @owned_scrolls[scroll_info.container] = scrolls_in_container
+	
+	  # Re-save user variables
+	  UserVars.owned_scrolls = @owned_scrolls
+	end
+
+	echo_message("SCROLL WAS GRABBED!")
+  end
+  
+  def full_name(noun)
+	noun_name = bput("tap #{noun}", '^You tap ((\w+\s*\W*)+) (that|inside)\s*.*')
+	noun_name =~ /^You tap ((\w+\s*\W*)+) (that|inside)\s*.*/i
+
+	if Regexp.last_match(3).include? ("inside")
+	  bput("get #{noun}", 'You get')
+	end
+	
+	name_full = Regexp.last_match(1).empty? ? "Undefined" : Regexp.last_match(1)
+	
+	return name_full
+  end
+
+  def get_scroll_info(index)
+	# Retrieve scroll info 
+	current_scrolls = UserVars.owned_scrolls
+	
+	the_scroll = nil
+	
+	current_scrolls.each do |container, scroll_array|
+	  scroll_array.each do |scroll_info|
```

Comment:
You could use `.find` instead of `.each` (see https://ruby-doc.org/core-2.2.0/Enumerable.html#method-i-find)